### ISSUE TITLE
Add support for dateStyle and timeStyle for Android

### ISSUE THREAD

**Issue by dlebedynskyi** [facebook/hermes#926]
## Problem

Intl on Android is still missing dateStyle and timeStyle. iOS already has it [since v0.12](https://github.com/facebook/hermes/blob/v0.12.0/lib/Platform/Intl/PlatformIntlApple.mm#L517).

## Solution

Ask is to add support for missing [Intl.DateTimeFormat API for Android](https://github.com/facebook/hermes/blob/main/doc/IntlAPIs.md)  [dateStyle/timeStyle](https://tc39.es/proposal-intl-datetime-style/).

## Additional Context

If possible, would be great to add rest of missing API 
* [dayPeriod](https://github.com/tc39/ecma402/issues/29)
* [fractionalSecondDigits](https://github.com/tc39/ecma402/pull/347)

to close rest of missing options on Intl.DateTimeFormat  as polyfill for Intl.DateTimeFormat is complex and heavy, and is currently required when dateStyle/timeStyle are used. 

Related https://github.com/facebook/hermes/issues/23#issuecomment-1446878656

**Comment by mganandraj** [comment]
I've created a PR : https://github.com/facebook/hermes/pull/927

@neildhar

**Comment by dlebedynskyi** [comment]
@mganandraj  sorry for late reply. 

We think there is a bug in how it is implemented. 

The way it is done, it does not match the spec - https://tc39.es/ecma402/#datetimeformat-objects. 

```
`If required is "date" or "any", then
a. For each property name prop of « "weekday", "year", "month", "day" », do
i. Let value be ? Get(options, prop).
ii. If value is not undefined, let needDefaults be false.
```

as example 
```
const date = new Date(Date.UTC(2012, 11, 20, 3, 0, 0, 200));
// request a weekday along with a long date
let options = {
  weekday: "narrow"
};
console.log(new Intl.DateTimeFormat("en-US", options).format(date)); 
// outputs "W, 12/19/2012" which is wrong,
```

this should just be "W" since using "weekday" negates the need for defaults.

Fix that we tested was here in https://github.com/facebook/hermes/blob/main/lib/VM/JSLib/Intl.cpp#L272

```diff
- {u"weekday", platform_intl::Option::Kind::String, 0},
+ {u"weekday", platform_intl::Option::Kind::String, kDateRequired},
```


We had tested your change prior to merge. So maybe we can do PR now. WDYT?

**Comment by mganandraj** [comment]
Hello @dlebedynskyi .. 100% agreed. This bug has existed from day 1 and somehow left unnoticed. We do run most relevant Test262 test cases which doesn't seem to cover this.

I'd appreciate if you can send a PR.

**Comment by nandorojo** [comment]
Hey all, would you recommend that we use #927 as a patch in our app? Or is there likelihood of this getting merged soon?

**Comment by neildhar** [comment]
Hey @nandorojo, it looks like this was already merged in a72115fb24414fde85e5fe5e29085431205454d0, and is present in RN 0.73. Closing this issue.
